app/components/signature-pill.js

```diff
@@ -42,13 +69,15 @@ export default class SignaturePillComponent extends Component {
       }
     }
 
+    this.schedulePublicationActivitiesRefresh();
+
     return {
       signingHubUrl,
       status,
     };
   });
 
-  data = trackedTask(this, this.loadData);
+  data = trackedTask(this, this.loadData, () => [this.triggerTask]); // Make the resource dependant on this.triggerTask
```

Comment:
I'm not sure I understand what's happening here.

So on line 72, on the first run of `this.loadData`, we load the first `status` and schedule the polling task. 
This will reload the `signFlow.status` after `SIGN_FLOW_STATUS_REFRESH_INTERVAL_MS` and set `this.triggerTask = new Date();`, which will trigger `this.loadData` because this trackedTask uses `this.triggerTask` as an argument ( https://ember-resources.pages.dev/funcs/util_ember_concurrency.trackedTask )

On line 43 in `this.loadData`, we then cancel the next `this.scheduledRefresh` to load the `status`, then we're back at line 72 and schedule the next task again.

So in other words, wouldn't it make more sense to put the reload code from lines 32 to 34 in `this.loadData` on lines 44 to 46? That way the scheduling task just updates the trigger for the trackedTask.
Or does this cause problems in Ember because you can't `reload()` on the first `await` of a relation?

Reply:
That also definitely works, I didn't want to change the original code too much. At that point we might be better off writing something like